### PR TITLE
Fix bug where multiple poll responses aren't displayed in aggregate

### PR DESCRIPTION
**Pull request by nikolas**
* Insert session key in poll response object, and tie that to the student so poll responses can be amended.

### CODE REVIEW

File: media/src/App.svelte [hunk 1]
```diff
@@ -448,17 +447,18 @@
...
+        } else if (data.message.pollResponse) {
+            const sessionKey = data.message.session_key;
+            let responseObj = {
+                choice: data.message.pollResponse,
+                session_key: sessionKey
+            };
+            // Remove any previous response from this session.
+            if (pollResponses.find(x => x.session_key === sessionKey)) {
+                pollResponses = pollResponses.filter(
+                    x => x.session_key !== sessionKey);
```
Comment (Evan-CTL) [inline media/src/App.svelte:459]: If you're doing a lookup using the session key, why not keep it as an object with `session_key` as the key and one of the object properties? It should be a faster lookup and you wouldn't have to delete the object in order to overwrite the response.

Regardless you don't need to filter out the object from the array either you can overwrite `x.choice`.

**Review by Evan-CTL — COMMENTED** [review]
Was switching out answers something that Drew had mentioned? I thought he wanted the to not be editable.

**Comment by nikolas** [comment]
Drew does want the answers to be editable and these changes build out the base for that. Next steps will be adding the edit UI for facilitating this.

File: media/src/App.svelte (hunk 1 above)
Comment (nikolas) [reply media/src/App.svelte:473]: Arrays are just nicer to deal with than objects and it's easier to process the data to get into d3. In this scenario I'm focusing on making the code simple and easier to understand - any performance optimizations can be added later if necessary, but it's not really an issue at this scale.

Comment (nikolas) [reply media/src/App.svelte:473]: But yeah - I did think of this alternative structure here.. hmm:

```
let pollResponses = {
  'session_key_1': 'a',
  'session_key_2': 'b'
};
```

Comment (nikolas) [reply media/src/App.svelte:473]: I'll do some adjustments here and see how that feels and get back to this. Converting this to a draft for now.

**Comment by nikolas** [comment]
Alright, I got this working with the pollResponses as an object and you're right, this makes more sense for now and it's much simpler to update.

Keep in mind these changes only take into account the multiple choice poll type - numeric poll will need some attention after these changes are in place.

**Pull request merged**